plugins/modules/luks_device.py

```diff
@@ -904,6 +1009,22 @@ def run_module():
     if module.params['label'] is not None and module.params['type'] == 'luks1':
         module.fail_json(msg='You cannot combine type luks1 with the label option.')
 
+    if module.params['remove_keyslot'] is not None:
+        if module.params['keyfile'] is None and module.params['passphrase'] is None:
+            module.fail_json(msg='Removing a keyslot requires the passphrase or keyfile of another slot')
+
+    luks_type = crypt.get_luks_type(conditions.get_device_name())
+    for parameter in ['keyslot', 'new_keyslot', 'remove_keyslot']:
+        if module.params[parameter] is not None and not validate_keyslot(module.params[parameter], luks_type):
+            if luks_type == 'luks1':
+                module.fail_json(msg='%s must be between 0 and 7 when using luks1' % (parameter))
+            else:
+                module.fail_json(msg='%s must be between 0 and 31 when using luks2' % (parameter))
```

Comment:
In this block, the test is off-loaded to the function `validate_keyslot`, but the same information - the max slot number depending on the LUKS type - is also needed here. How about changing `validate_keyslots` to call `module.fail_json` directly?